### PR TITLE
feat(api): Allow invite requests to be approved

### PR DESCRIPTION
**Pull request by maheskett**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_invite_request_details.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize, OrganizationMemberWithTeamsSerializer
+from sentry.api.serializers.rest_framework import ListField
+from sentry.models import AuditLogEntryEvent, InviteStatus, OrganizationMember, Team, TeamStatus
+from sentry.signals import member_invited
+
+from .organization_member_index import save_team_assignments
+
+
+class InviteRequestSerializer(serializers.Serializer):
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_invite_request_details.py]: Can we reuse the other validator here somehow? Ideally not subclassing.

Comment (maheskett) [reply src/sentry/api/endpoints/organization_invite_request_details.py]: updated

File: src/sentry/api/endpoints/organization_invite_request_details.py [hunk 2]
```diff
@@ -1,14 +1,202 @@
...
+            and not request.access.has_scope("member:admin")
+            and not request.access.has_scope("member:write")
+        ):
+            return Response(
+                {"detail": ERR_INSUFFICIENT_ROLE_UPDATE}, status=status.HTTP_403_FORBIDDEN
+            )
+
+        serializer = OrganizationMemberSerializer(
+            data=request.data,
+            context={"organization": organization, "allowed_roles": roles.get_all()},
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_invite_request_details.py]: allowed roles should probably be the users allowed roles, so this should be closer to what the member invite endpoint.

File: src/sentry/api/endpoints/organization_invite_request_details.py [hunk 3]
```diff
@@ -1,14 +1,202 @@
...
+                context={"request": request, "organization": organization, "member": member},
+            )
+
+            if not serializer.is_valid():
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+            result = serializer.validated_data
+
+            if result.get("approve"):
+                if not member.invite_approved:
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_invite_request_details.py]: We can condense this I think `if result.get("approve") and not member.invite_approved:`

**Review by evanpurkhiser — CHANGES_REQUESTED** [review]
This is fantastic, just a few minor changes.

Primarily, we need to make sure they can't invite a role that is greater than their role.

File: src/sentry/api/endpoints/organization_invite_request_details.py (hunk 2 above)
Comment (maheskett) [reply src/sentry/api/endpoints/organization_invite_request_details.py]: updated. only added this check to the approve serializer, so members can still update invite requests regardless of the role.

File: src/sentry/api/endpoints/organization_invite_request_details.py [hunk 4]
```diff
@@ -1,14 +1,210 @@
...
+        ):
+            raise serializers.ValidationError(ERR_JOIN_REQUESTS_DISABLED)
+
+        if not request.access.has_scope("member:admin") and not request.access.has_scope(
+            "member:write"
+        ):
+            raise serializers.ValidationError(ERR_INSUFFICIENT_SCOPE)
+
+        # members cannot invite roles higher than their own
+        if member.role not in {r.id for r in allowed_roles}:
```
Comment (EvanPurkhiser) [inline src/sentry/api/endpoints/organization_invite_request_details.py]: Why do you have to validate this here? Don't we already validate that elsewhere?

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**